PowerShellTools/Classification/PowerShellBraceMatchingTagger.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+using PowerShellTools.Intellisense;
+
+namespace PowerShellTools.Classification
+{
+    internal sealed class PowerShellBraceMatchingTagger : ITagger<TextMarkerTag>, INotifyBufferUpdated
+    {
+	ITextView _textView;
+	ITextBuffer _textBuffer;
+	SnapshotPoint? _caretPos;
+	private IDictionary<int, int> _startBraces;
+	private IDictionary<int, int> _endBraces;
+	private const string MarkedColor = "blue";

```

Comment:
Good catch. I tried this on high-contrast themes, not obvious. May need your help on how turn this into different colors based on theme mode.